File: C++/Medium/WrongAnswer/find-all-lonely-numbers-in-the-array_1.cpp
class Solution {
public:
    vector<int> findLonely(vector<int>& nums) {
        
        vector<int> res;
        int n = nums.size();
        if (n == 1) {
            res.push_back(nums[0]);
            return res;
        }
        if (nums[0] != nums[1]) {
            res.push_back(nums[0]);
        }
        for (int i = 1; i < n - 1; i++) {
            if (nums[i] != nums[i - 1] && nums[i] != nums[i + 1]) {
                res.push_back(nums[i]);
            }
        }
        if (nums[n - 1] != nums[n - 2]) {
            res.push_back(nums[n - 1]);
        }
        return res;
    }
};

File: Java/Medium/WrongAnswer/find-all-lonely-numbers-in-the-array_1.java
class Solution {
    public List<Integer> findLonely(int[] nums) {
        
        List<Integer> result = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            if (i == 0 && nums[i] != nums[i + 1]) {
                result.add(nums[i]);
            } else if (i == nums.length - 1 && nums[i] != nums[i - 1]) {
                result.add(nums[i]);
            } else if (i != 0 && i != nums.length - 1 && nums[i] != nums[i - 1] && nums[i] != nums[i + 1]) {
                result.add(nums[i]);
            }
        }
        return result;
    }
}